### DISCUSSION TITLE
Datasets retrieval testing through Knowledge API

### DISCUSSION THREAD

**Discussion by infinitr0us** [langgenius/dify#3992] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I was trying to use the Knowledge API to manipulate the datasets that I created. However, I found that the current Knowledge API does not support retrieval testing of a certain dataset. 

### 2. Describe the feature you'd like to see

Support retrieval testing for a certain dataset in Knowledge API.

### 3. How will this feature improve your workflow or experience?

This will be beneficial to custom knowledge building, through large amount of retrieval testing with API calling, I could collect and analysis the dataset and get better idea on how to improve the presentation and custom settings of it.

### 4. Additional context or comments

Thanks for your great work on building Dify, especially the API system inside. I agree that the current Knowledge API is already pretty good. Just need to a few more polishing for a little bit more functions.

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by HoekWax** [comment]
I am facing the same use case, it would be great to be able retrieve documents in a specific dataset based on a prompt/query.

Either allowing this from the API or being able to create a workflow, adding a retrieval node and being able to pass a variable which is the datasetId (instead of having to set the dataset from the UI) then getting an answer
